kanidmd/src/lib/tracing_tree/formatter.rs

```diff
@@ -0,0 +1,285 @@
+use super::subscriber::{EventTagSet, TreeEvent, TreeProcessed, TreeSpanProcessed};
+use serde::{ser::SerializeStruct, Serialize};
+use std::fmt;
+use std::io::{self, Write as _};
+use tracing::Level;
+use tracing_serde::AsSerde;
+
+#[derive(Clone, Copy, Debug)]
+pub enum LogFmt {
+    Json,
+    Pretty,
+}
+
+const EVENT_UUID: &str = "00000000-0000-0000-0000-000000000000";
```

Comment:
Const's should probably be in that place where @Firstyear documents past decisions about numbers, and I'm not sure the default should be all-0's - isn't that the Anonymous user uuid?

Reply:
I'm not sure here, the all 0000 is pretty clear, but it is also likely to cause confusion if someone is grepping for anonymous. https://github.com/kanidm/kanidm/tree/master/kanidmd/src/lib/constants